draft-ietf-tls-tls13.md

```diff
@@ -2649,16 +2651,25 @@ Structure of this message:
        } Certificate;
 
 certificate_list
-: This is a sequence (chain) of certificates.  The server's
-  certificate MUST come first in the list.  Each following
-  certificate MUST directly certify the one preceding it.  Because
-  certificate validation requires that root keys be distributed
-  independently, the self-signed certificate that specifies the root
-  certification authority (CA) MAY be omitted from the chain, under the
-  assumption that the client must already possess it in order to
-  validate it in any case.
+: This is a sequence (chain) of certificates. The sender's
+  certificate MUST come first in the list. Each following
+  certificate SHOULD directly certify one preceding it. Because
+  certificate validation requires that trust anchors be distributed
+  independently, a self-signed certificate that specifies a
+  trust anchor MAY be omitted from the chain, provided that
+  supported peers are known to possess any omitted certificates
+  they may require.
 {:br }
 
+Note: Prior to TLS 1.3, "certificate_list" ordering was required to be strict,
+however some implementations already allowed for some flexibility. For maximum
+compatibility, all implementations SHOULD be prepared to handle potentially
+extraneous certificates and arbitrary orderings from any TLS version (with
+the exception of the sender's certificate, which is expected to be first). Some
```

Comment:
My rationale for using the word "expected" here is that there is already a "MUST" above for this and from the client's perspective, whilst the spec says it MUST be first, it cannot safely assume so and should at least be prepared to abort in the event that it is not.

What might be a better route is to drop the "expected" part in the parenthetical and add a requirement that a client "MUST" abort if it's not first. Probably always better to be more explicit, even if a little more wordy. I'll push another commit to this PR with a change in a bit.